### ISSUE TITLE
Simplify API and examples

### ISSUE THREAD

**Issue by danrubel** [danrubel/rpi_gpio.dart#5]
- [ ] Provide top level `gpio` accessor for obtaining the initialized Gpio instance for interacting with the GPIO pins running on the Raspberry Pi.
- [x] Provide top level `pin` accessor for obtaining the initialized Gpio pin instances
- [ ] Provide a `new Gpio(hardware)` constructor for mocking
- [ ] Deprecate the `Gpio.hardware =` setter in 0.2.x and remove it in subsequent release
- [x] Remove mocking from the existing examples and provide one new example with mocking
- [ ] Move hardware API to dart:pi that is delivered as part of an [sdk extension](https://github.com/dart-lang/sdk/issues/23583) - see [Simplify installation with prebuilt SDK with library](https://github.com/danrubel/rpi_gpio.dart/issues/6)

**Comment by danrubel** [comment]
For example, it should be easy to consume te the rpi_gpio package in order to drive hardware buttons and strips of LEDs.

``` dart
import 'dart:math';
import 'package:rpi_gpio/rpi_gpio.dart';

Random random = new Random();
double next() => random.nextDouble();

void change(Strip led) {
  led
    ..red = next()
    ..blue = next()
    ..green = next();
}

main() {
  var led = new Strip(pin(1, output), pin(2, output), pin(3, output));
  change(led);
  new Button(pin(4, input), () => change(led));
}
```

**Comment by danrubel** [comment]
New API published in version 0.5.0
